### ISSUE TITLE
[Java] Implement new Concept Exercise: constructors

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1864]
This issue describes how to implement the `constructors` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of the exercise is to learn how to use constructor methods to instantiate classes

## Learning objectives

- Know what constructors are
- Know how to define parameterless constructors
- Know how to define parameterized constructors

## Out of scope

- Constructor overloading
- Private constructors
- Static constructors

## Concepts

- `constructors`: know what constructors are; know how to define parameterless constructors; know how to define parameterized constructors; know how to use constructor overloading; know how to define private constructors.

## Prequisites

- `classes`: know how to work with classes.
- `numbers`: know how compare numbers.
- `conditionals`: know how to do conditional logic.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer)

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer)

## Implementing
To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by ystromm** [comment]
Looks reasonable to port the c# version. What do you think?

**Comment by mirkoperillo** [comment]
@ystromm absolutely yes.
 Let me know if you can be interested in implement this

**Comment by ystromm** [comment]
Sure I can give it a try!

**Comment by mirkoperillo** [comment]
Fixed by #2255
